core/src/test/java/org/apache/accumulo/core/file/rfile/GenerateSplitsTest.java

```diff
@@ -0,0 +1,148 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.file.rfile;
+
+import static org.apache.accumulo.core.file.rfile.GenerateSplits.main;
+import static org.apache.accumulo.core.file.rfile.RFileTest.newColFamByteSequence;
+import static org.apache.accumulo.core.file.rfile.RFileTest.newKey;
+import static org.apache.accumulo.core.file.rfile.RFileTest.newValue;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.List;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+@SuppressFBWarnings(value = "PATH_TRAVERSAL_IN", justification = "paths provided by test")
+public class GenerateSplitsTest {
+  private static final Logger log = LoggerFactory.getLogger(GenerateSplitsTest.class);
+
+  @ClassRule
+  public static final TemporaryFolder tempFolder =
+      new TemporaryFolder(new File(System.getProperty("user.dir") + "/target"));
+
+  private static final RFileTest.TestRFile trf = new RFileTest.TestRFile(null);
+  private static String fileName;
+
+  /**
+   * Creates a test file with 84 bytes of data and 2 Locality groups.
+   */
+  @BeforeClass
+  public static void createFile() throws IOException {
+    trf.openWriter(false);
+    trf.writer.startNewLocalityGroup("lg1", newColFamByteSequence("cf1", "cf2"));
+    trf.writer.append(newKey("r1", "cf1", "cq1", "L1", 55), newValue("foo1"));
+    trf.writer.append(newKey("r2", "cf2", "cq1", "L1", 55), newValue("foo2"));
+    trf.writer.append(newKey("r3", "cf2", "cq1", "L1", 55), newValue("foo3"));
+    trf.writer.startNewLocalityGroup("lg2", newColFamByteSequence("cf3", "cf4"));
+    trf.writer.append(newKey("r4", "cf3", "cq1", "L1", 55), newValue("foo4"));
+    trf.writer.append(newKey("r5", "cf4", "cq1", "L1", 55), newValue("foo5"));
+    trf.writer.append(newKey("r6", "cf4", "cq1", "L1", 55), newValue("foo6"));
+    trf.closeWriter();
+
+    File file = tempFolder.newFile("testGenerateSplits.rf");
+    FileOutputStream fileOutputStream = new FileOutputStream(file);
+    fileOutputStream.write(trf.baos.toByteArray());
+    fileOutputStream.flush();
+    fileOutputStream.close();
+
+    fileName = file.getAbsolutePath();
+    log.info("Wrote to file {}", fileName);
+  }
+
+  @AfterClass
+  public static void cleanUp() {
+    File file = new File(fileName);
+    if (file.delete())
+      log.info("Cleaned up test file {}", fileName);
```

Comment:
Can usually do something like this in a one-liner, since either case are acceptable, and we probably don't need the spammy log message. We just need to know if something unexpected happened, which the following will do.

```suggestion
    assertTrue(file.delete() || !file.exists());
```

However, I'm thinking this entire cleanup method isn't necessary at all. The file is created inside the temporary folder. By definition, it will be cleaned up when that folder is cleaned up after the tests.

Reply:
I had added this when i created the test initially because I was unsure but I think you are right. I thought I saw another Test doing this as well but can't find it now.

Reply:
Removed in 4d1fdc6